be/src/exec/vectorized/hdfs_scanner_orc.cpp

```diff
@@ -70,16 +71,25 @@ class ORCHdfsFileStream : public orc::InputStream {
             size_t idx = offset - _cache_offset;
             memcpy(buf, _cache_buffer.data() + idx, length);
         } else {
-            doRead(buf, length, offset);
+            doRead(buf, length, offset, false);
         }
     }
 
-    void doRead(void* buf, uint64_t length, uint64_t offset) {
+    void doRead(void* buf, uint64_t length, uint64_t offset, bool direct) {
         if (buf == nullptr) {
             throw orc::ParseError("Buffer is null");
         }
-
-        Status status = _file->read_at_fully(offset, buf, length);
+        Status status;
+        if (direct || !_buffer_stream_enabled) {
+            status = _file->read_at_fully(offset, buf, length);
+        } else {
+            const uint8_t* ptr = nullptr;
+            size_t nbytes = length;
+            status = _buffer_stream.get_bytes(&ptr, offset, &nbytes);
+            if (status.ok()) {
+                ::memcpy(buf, ptr, length);
```

Comment:
length -> nbytes ?

Reply:
they are the same value.

Reply:
But from from the view of the caller, only the 'nbytes' is the real size returned by stream and we may not depend on current implementation which makes them equal.

Reply:
OK, maybe we can check `nbytes == length` here. Because we are expected to read length bytes, otherwise there is error. Do you agree?

Reply:
Yes，that is safer.